packages/next/src/build/index.ts

```diff
@@ -2258,11 +2287,17 @@ export default async function build(
               appConfig.revalidate === 0 ||
               exportResult.byPath.get(page)?.revalidate === 0
 
-            if (hasDynamicData && pageInfos.get(page)?.isStatic) {
+            const pageInfo = pageInfos.get(page)
+            if (!pageInfo) {
+              throw new Error(
+                `Invariant: page info for ${page} is missing from registry`
+              )
+            }
+
+            if (hasDynamicData && pageInfo.isStatic) {
               // if the page was marked as being static, but it contains dynamic data
               // (ie, in the case of a static generation bailout), then it should be marked dynamic
-              pageInfos.set(page, {
-                ...(pageInfos.get(page) as PageInfo),
+              patchPageInfos(pageInfos, page, {
```

Comment:
You just existence checked the pageInfo above. Consider whether the patching needs to happen at all (and just use mutation to "patch" the values). If patching is necessary consider changing the API to take the current pageInfo object rather than a path. then just `.set` it.

there are other places where you use this patch function but you can extract the `get` into the outer scope and throw there if it is missing.